src/result/result.py

```diff
@@ -202,6 +202,19 @@ def or_else(self, op: object) -> Ok[T]:
         """
         return self
 
+    def inspect(self, op: Callable[[T], None]) -> Result[T, E]:
```

Comment:
Change all of the `op` types from returning `None` to `Any`, because we shouldn't unnecessarily restrict the user from using a functions that can also return values, we can just ignore any returned value.

```suggestion
    def inspect(self, op: Callable[[T], Any]) -> Result[T, E]:
```

Reply:
Yeah, nice take!